src/modules/mavlink/streams/GIMBAL_DEVICE_INFORMATION.hpp

```diff
@@ -0,0 +1,96 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef GIMBAL_DEVICE_INFORMATION_HPP
+#define GIMBAL_DEVICE_INFORMATION_HPP
+
+#include <uORB/topics/gimbal_device_information.h>
+
+class MavlinkStreamGimbalDeviceInformation : public MavlinkStream
```

Comment:
This should not be required, unless you implement the case where PX4 is the gimbal device but not the manager?

Instead the gimbal manager should forward the message or send the gimbal manager information. Wouldn't that be enough?

Reply:
Don't we plan to let PWM gimbals appear as mavlink v2 gimbal devices? In this case this PR could be helpful.

Reply:
> Don't we plan to let PWM gimbals appear as mavlink v2 gimbal devices? In this case this PR could be helpful.

@dayjaby we changed our mind on that again because it didn't look like ArduPilot or PX4 are likely to implement it that way.

Reply:
> I am a bit rusty on the mavlink forwarding side in PX4, you mean that we could simply forward the request to the gimbal and then re-forward the answer to the GCS without going through the uORB population?
In that case could provide some pointers?

Right. By just enabling forwarding, commands addressed to the gimbal device component ID would get forwarded and answered automatically.

> Are you referring to what we discussed a few comments above, i.e. extend gimbal manager information to include the gimbal device vendor, model, ...?

I see the problem. I guess in the case that PX4 has "its own" gimbal, e.g. a PWM one, it would have to fake the gimbal device message as you suggest in this PR.